tests/test_py37.py

```diff
@@ -111,9 +111,9 @@ class Foo(BaseModel):
     }
 
     with pytest.raises(ValidationError) as exc_info:
-        module.Foo(b={'a': '321'}, c=[{'b': 234}], d={'bar': {'a': 345}})
```

Comment:
There's a caveat here: for the inner `b`, it would try to use `234` as an arbitrary class instance.

It would then try to extract each of the fields of the self-referencing `Foo` model.

As `234` is just a number, it wouldn't have any of the fields.

But, as all the fields are optional or with defaults, the final value of `b` would be just a `Foo` model with all the default values.

So, in a similar way that an `int` would be converted to `True` when a field has type `bool`, any arbitrary object (of any value, including `int`) without attributes would be treated as an empty `dict`, and converted to the equivalent of a model with default values.

---

If you don't like it @samuelcolvin , other option I see would be to accept an object as an arbitrary only if it is not a sub-class of the known singular classes (`int`, `str`, etc).